src/Orleans/Serialization/BuiltInTypes.cs

```diff
@@ -1275,6 +1278,34 @@ internal static object CopyType(object obj)
         }
 
         #endregion Type
+    
+        #region CancellationTokenWrapper
+        internal static void SerializeCancellationTokenWrapper(object obj, BinaryTokenStreamWriter stream, Type expected)
+        {
+            var ctw = (CancellationTokenWrapper)obj;
+            var cancelled = ctw.CancellationToken.IsCancellationRequested;
+            if (!cancelled)
+            {
+                CancellationTokenManager.RegisterTokenCallbacks(ctw);
```

Comment:
Initially this was made with `Action` callaback to avoid unnecessary computations in case of local grain call, and then replaced due to https://github.com/dotnet/orleans/pull/1569#discussion_r56413949. But if you think that now the cost of grain extension call and `CancellationTokenSource` allocation is neglectible compared to the code readability & supportability I will move it to `SendRequest` methods.